ra2ce/analyses/analysis_config_wrapper.py

```diff
@@ -0,0 +1,185 @@
+        return filepath.parent.parent
+
+    @staticmethod
+    def get_data_output(ini_file: Path) -> Optional[Path]:
+        _root_path = AnalysisConfigWrapper.get_network_root_dir(ini_file)
+        _project_name = ini_file.parent.name
+        return _root_path.joinpath(_project_name, "output")
+
+    @property
+    def root_dir(self) -> Path:
+        return self.get_network_root_dir(self.ini_file)
+
+    def initialize_output_dirs(self) -> None:
+        """
+        #Initializes the required output directories for a Ra2ce analysis.
+        """
+        # Create the output folders
+        for a in self.config_data.analyses:
+            output_path = self.config_data.output_path.joinpath(a.analysis)
+            output_path.mkdir(parents=True, exist_ok=True)
+
+    @classmethod
+    def from_data(
+        cls, ini_file: Path, config_data: AnalysisConfigData
+    ) -> AnalysisConfigWrapper:
+        """
+        Initializes an `AnalysisConfigWrapper` with the given parameters.
+
+        Args:
+            ini_file (Path): Path to the ini file containing the analysis data.
+            config_data (AnalysisIniConfigData): Ini data representation.
+
+        Raises:
+            FileNotFoundError: When the provided `ini file` cannot be found.
+
+        Returns:
+            AnalysisConfigWrapper: Initialized instance.
+        """
+        if not ini_file.is_file():
+            raise FileNotFoundError(ini_file)
+        _new_analysis = cls()
+        _new_analysis.ini_file = ini_file
+        _new_analysis.config_data = config_data
+        return _new_analysis
+
+    @classmethod
+    def from_data_with_network(
+        cls,
+        ini_file: Path,
+        config_data: AnalysisConfigData,
+        network_config: NetworkConfigWrapper,
+    ) -> AnalysisConfigWrapper:
+        """
+        Initializes an `AnalysisConfigWrapper` with the given parameters,
+        with network_config as input from the network phase.
+
+        Args:
+            ini_file (Path): Path to the ini file containing the analysis data.
+            config_data (AnalysisIniConfigData): Ini data representation.
+            network_config (NetworkConfig): Network configuration to be used on this analysis.
+
+        Returns:
+            AnalysisConfigWrapper: Initialized instance.
+        """
+        _new_analysis = cls.from_data(ini_file, config_data)
+        _new_analysis.config_data.files = network_config.files
+        _new_analysis.config_data.network = network_config.config_data.network
+        _new_analysis.config_data.origins_destinations = (
+            network_config.config_data.origins_destinations
+        )
+        # Graphs are retrieved from the already configured object
+        _new_analysis.graphs = network_config.graphs
+
+        return _new_analysis
+
+    @classmethod
+    def from_data_without_network(
+        cls,
+        ini_file: Path,
+        config_data: AnalysisConfigData,
+    ) -> AnalysisConfigWrapper:
```

Comment:
Would it make sense to get some of this logic into the factory so that here we only have `from_data_with_network`? Let's aim to move all this logic to the factory so the responsibility to initialize the wrappers lays there.

Reply:
Done